### PR TITLE
Release notes for V0.8

### PR DESCRIPTION
**Pull request by SueChaplain**
Create directory structure and draft markdown file
to support the Eclipse OpenJ9 V0.8 release.

Issue #1052

Signed-off-by: Sue Chaplain <sue_chaplain@uk.ibm.com>

### CODE REVIEW

**Comment by DanHeidinga** [comment]
@SueChaplain Should the release notes be included in the repo or is adding them directly to the "release" sufficient?  See https://help.github.com/articles/creating-releases/ for details on how the markdown can be associated directly to the release tag.

**Comment by DanHeidinga** [comment]
If we're going to include this in the repo (see previous comment for other options), then it should be a subdirectory under the `doc` directory, not at the top-level.

File: release-notes/0.8/0.8.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Eclipse OpenJ9 version 0.8 release notes
+
+## Supported platforms
+
+JCK compliance status for **OpenJDK with Eclipse OpenJ9** binaries:
```
Comment (DanHeidinga) [inline release-notes/0.8/0.8.md]: This statement on JCK compliance should reference the AdoptOpenJDK builds of this release have passed (or not) JCKs.  OpenJ9 doesn't have a JCK license and is relying on Adopt to JCK the OpenJDK with Eclipse OpenJ9 binaries.

File: release-notes/0.8/0.8.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+- Java 9 compliant
+- Java 8 expected to be compliant
```
Comment (DanHeidinga) [inline release-notes/0.8/0.8.md]: The focus at Adopt has been on running the Java 8 JCKs.  As the "non-production" use banner is still up at Adopt and won't come down until binaries pass the JCKs, we can't claim compliance for either 8 or 9.

File: release-notes/0.8/0.8.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+#### Direct dump reader
+[Issue #378](https://github.com/eclipse/openj9/issues/378):
+The direct dump reader is not yet working, which has an impact on diagnosing problems with the VM, Garbage Collector, and the JIT. For example, [issue #968](https://github.com/eclipse/openj9/issues/968).
+
+#### Signal handling
+[Issue #54](https://github.com/eclipse/openj9/issues/54): Some signals are not being handled correctly by the VM. In particular, for CTRL-C or other signals that shutdown the VM, the shutdown hooks are not running. See [Issue #378](https://github.com/eclipse/openj9/issues/378)
+
+#### Large Java heaps
+[Issue #479](https://github.com/eclipse/openj9/issues/479): Java heaps bigger than 57 GB (or ~62 GB with command line options) are not supported.
```
Comment (DanHeidinga) [inline release-notes/0.8/0.8.md]: The builds from Adopt only build the "compressedrefs" binary and thus have this limitation.  OpenJ9 supports building both the compressed and non-compressed versions.  Eventually we'll enable both in the same binary, either by composing the two packages or by enabling a single binary to do both based on runtime checks.

**Comment by SueChaplain** [comment]
_"@SueChaplain Should the release notes be included in the repo or is adding them directly to the "release" sufficient? See https://help.github.com/articles/creating-releases/ for details on how the markdown can be associated directly to the release tag."_

I followed the link but it wasn't completely clear to me how the release notes are embedded in the release information. I created a markdown file for the repo because I thought that was the conclusion from our initial discussion in #1052. You pointed to https://github.com/dotnet/coreclr/releases, where they have a clean, simple set of release information that links to a set of markdown files in their repo. 

Right now, we don't have many issues/limitations to list but they would still clog up that release page. In the future I am guessing that we'll want to document new features or behaviour changes that might break people. The user docs will want to point to those too but it looks like all we'll be able to point to is the release tag. E.g. https://github.com/dotnet/coreclr/releases/tag/v1.1.6 . 

Having a markdown file enables us to point to specific parts within a markdown file. For example, this link takes you to a specific section of the build instructions: https://github.com/eclipse/openj9/blob/master/buildenv/Build_Instructions_V8.md#aix

So I think that having an MD file is the right way to go and I'm happy to move the structure under /docs . As an aside, I'm not sure what other content you intend to store in /docs versus the user documentation that we'll store in a separate repo and host online or any API documentation when that arrives. We should have a clear content strategy, which I'm happy to help with.

File: release-notes/0.8/0.8.md (hunk 1 above)
Comment (SueChaplain) [reply release-notes/0.8/0.8.md]: For now I've removed references to JCK compliance completely and made mention of testing against the OpenJDK test suite.

File: release-notes/0.8/0.8.md (hunk 2 above)
Comment (SueChaplain) [reply release-notes/0.8/0.8.md]: Handled, as above.

File: release-notes/0.8/0.8.md (hunk 3 above)
Comment (SueChaplain) [reply release-notes/0.8/0.8.md]: Added additional explanation to the Large Java Heap item.

**Comment by DanHeidinga** [comment]
Thanks for taking a look at the options and making such an easy to agree with case for putting the files in /doc/release-notes.  I'm onboard with this approach.

> I'm not sure what other content you intend to store in /docs

This has tended to be developer-centric info about OpenJ9 and how it works.  User-centric info should definitely be hosted as part of the docs repo.

File: doc/release-notes/0.8/0.8.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Eclipse OpenJ9 version 0.8 release notes
+
+## Supported platforms
+
+OpenJDK binaries that contain the Eclipse OpenJ9 virtual machine are available from the
+[AdoptOpenJDK community](https://adoptopenjdk.net/?variant=openjdk8-openj9). These builds are
+being tested against the OpenJDK test suite.
```
Comment (DanHeidinga) [inline doc/release-notes/0.8/0.8.md]: @smlambert Can you provide some details on the levels of testing for OpenJ9?  I know there's a mix of FV, SV, and OpenJDK tests being run at one or both of the OpenJ9 and Adopt Jenkins.  A little more detail for the release notes would help.

File: doc/release-notes/0.8/0.8.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+#### Direct dump reader
+[Issue #378](https://github.com/eclipse/openj9/issues/378):
+The direct dump reader is currently not working, which has an impact on diagnosing problems with the VM, Garbage Collector, and the JIT. For example, [issue #968](https://github.com/eclipse/openj9/issues/968).
+
+#### Signal handling
+[Issue #54](https://github.com/eclipse/openj9/issues/54): Some signals are not being handled correctly by the VM. In particular, for CTRL-C or other signals that shutdown the VM, the shutdown hooks are not running. See [Issue #378](https://github.com/eclipse/openj9/issues/378)
+
+#### Large Java heaps
+[Issue #479](https://github.com/eclipse/openj9/issues/479): The binaries available at AdoptOpenJDK are built only with the compressed references VM, which does not support Java heaps larger
+than 57 GB (or ~62 GB with command line options). When builds that contain the non-compressed references VM are available, this limitation will be removed.
```
Comment (DanHeidinga) [inline doc/release-notes/0.8/0.8.md]: Can you add a note along the lines of "Users are able to build the non-compressed refs VM themselves."

File: doc/release-notes/0.8/0.8.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+#### Graphics Processor Unit (GPU) support
+GPU acceleration with `java.util.Arrays.sort()` is not supported, although `com.ibm.gpu.Maths` can be used directly.
+
+#### Functional verification tests
+The following issues relate to test failures:
+
+- [#238](https://github.com/eclipse/openj9/issues/238), [#252](https://github.com/eclipse/openj9/issues/252), [#244](https://github.com/eclipse/openj9/issues/244), [#362](https://github.com/eclipse/openj9/issues/362), [#366](https://github.com/eclipse/openj9/issues/366), [#480](https://github.com/eclipse/openj9/issues/480), [#1128](https://github.com/eclipse/openj9/issues/1128).
+
+Because some tests are not yet working, functional problems might not be exposed. In addition, some IBM authored functional verification testing is not yet contributed to Eclipse, such as #724.
```
Comment (DanHeidinga) [inline doc/release-notes/0.8/0.8.md]: Should `#724` be a link?

Comment (SueChaplain) [reply doc/release-notes/0.8/0.8.md]: Yep, missed it! Now fixed.

File: doc/release-notes/0.8/0.8.md (hunk 5 above)
Comment (SueChaplain) [reply doc/release-notes/0.8/0.8.md]: Done

**Comment by DanHeidinga** [comment]
@smlambert When do you think you'll be able to review the testing section of the release notes?  We're getting close to being ready for the 0.8 release so there's some urgency to this :)

File: doc/release-notes/0.8/0.8.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+#### Large Java heaps
+[Issue #479](https://github.com/eclipse/openj9/issues/479): The binaries available at AdoptOpenJDK are built only with the compressed references VM, which does not support Java heaps larger
+than 57 GB (or ~62 GB with command line options). When builds that contain the non-compressed references VM are available, this limitation will be removed. If
+you want to build a non-compressed references VM yourself, you can do so now by following our [detailed build instructions](https://github.com/eclipse/openj9/blob/master/buildenv/Build_Instructions_V8.md).
+
+#### Graphics Processor Unit (GPU) support
+GPU acceleration with `java.util.Arrays.sort()` is not supported, although `com.ibm.gpu.Maths` can be used directly.
+
+#### Functional verification tests
```
Comment (pshipton) [inline doc/release-notes/0.8/0.8.md]: I'm wondering if we should have this section. More issues have been reported since this list was created. Not necessarily from the functional verification test bucket, but from additional testing being running. It might make more sense to provide a link like https://github.com/eclipse/openj9/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22test+failure%22+-label%3Ajdk9+-label%3Ajdk10+, although this link is only good at the time when we branch for the 0.8 release.

Comment (smlambert) [reply doc/release-notes/0.8/0.8.md]: Reviewing now, thanks for the 2nd note, I somehow missed the first request on this one.

File: doc/release-notes/0.8/0.8.md (hunk 4 above)
Comment (smlambert) [reply doc/release-notes/0.8/0.8.md]: These builds are being tested against several test suites, including the OpenJDK regression tests, the AdoptOpenJDK and OpenJ9 System Verification (SV) tests, the OpenJ9 Functional Verification (FV) tests, and the Java Compatibility Kit (JCK) tests.

File: doc/release-notes/0.8/0.8.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+The following issues relate to test failures:
+
+- [#238](https://github.com/eclipse/openj9/issues/238), [#252](https://github.com/eclipse/openj9/issues/252), [#244](https://github.com/eclipse/openj9/issues/244), [#362](https://github.com/eclipse/openj9/issues/362), [#366](https://github.com/eclipse/openj9/issues/366), [#480](https://github.com/eclipse/openj9/issues/480), [#1128](https://github.com/eclipse/openj9/issues/1128).
+
+Because some tests are not yet working, functional problems might not be exposed. In addition, some IBM authored functional verification testing is not yet contributed to Eclipse, such as issue  [#724](https://github.com/eclipse/openj9/issues/724).
```
Comment (smlambert) [inline doc/release-notes/0.8/0.8.md]: I personally would reword this as "Because some tests are not yet enabled" rather than not yet working, if you are referring to the tests not yet migrated to the OpenJ9 repo.  As for the next sentence, is it true that we have traceformat tests that have not been contributed?  We will have to track it down, as tests for utilities was not owned by our team previously.

File: doc/release-notes/0.8/0.8.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+#### API documentation
+
+Some OpenJ9 extensions are missing, see [issue #897](https://github.com/eclipse/openj9/issues/897).
+
+Some OpenJ9 extensions are unsupported, see [issue #898](https://github.com/eclipse/openj9/issues/898).
+
+#### Shared libararies
```
Comment (smlambert) [inline doc/release-notes/0.8/0.8.md]: libraries vs libararies

File: doc/release-notes/0.8/0.8.md (hunk 4 above)
Comment (smlambert) [reply doc/release-notes/0.8/0.8.md]: Note: We are additionally running some 3rd party application testing (a set that will grow by our next release) and some perf benchmarks, but we are not gating releases (https://github.com/AdoptOpenJDK/openjdk-tests/issues/186) on these two categories of tests.

File: doc/release-notes/0.8/0.8.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+[AdoptOpenJDK community](https://adoptopenjdk.net/?variant=openjdk8-openj9). These builds are
+being tested against the OpenJDK test suite.
+
+The following platforms are available:
+
+|Operating system | Platform architecture | Comments                                    |
+|-----------------|-----------------------|---------------------------------------------|
+|Linux            | x86, IBM Power, IBM Z | Windows subsytem for Linux is not supported |
+|AIX              | IBM Power             | Untested                                    |
+|Windows          | x86                   | Untested                                    |
```
Comment (smlambert) [inline doc/release-notes/0.8/0.8.md]: We can have a Windows test build at AdoptOpenJDK, there is a machine configuration problem however, currently only a single run of the OpenJDK regression tests can run cleanly before a reboot is needed, 2nd run fails with some permission issues.  So, we can report on open test results for the Windows platform if needed.

Comment (pshipton) [reply doc/release-notes/0.8/0.8.md]: for Java 8? I don't see any Adopt Windows builds for Java 8 yet. This release is only supporting Java 8.

Comment (pshipton) [reply doc/release-notes/0.8/0.8.md]: Speaking of which, the release notes don't mention this at all.

Comment (smlambert) [reply doc/release-notes/0.8/0.8.md]: Because of the machine issue, the windows test jobs were moved to the "work in progress" view:

However, I see we only have the hotspot run: https://ci.adoptopenjdk.net/view/work%20in%20progress/job/openjdk_test_x86-64_windows/250/  

as there is no successful openj9 builds at AdoptOpenJDK at the moment to trigger the tests for the openj9 test build:
https://ci.adoptopenjdk.net/view/work%20in%20progress/job/openjdk8_j9_openjdktest_x86-64_windows/

**Comment by DanHeidinga** [comment]
@SueChaplain Can you also add a link in the release notes to the Eclipse release plan: https://projects.eclipse.org/projects/technology.openj9/releases/0.8/plan

File: doc/release-notes/0.8/0.8.md (hunk 10 above)
Comment (smlambert) [reply doc/release-notes/0.8/0.8.md]: Other links: in case you wish to reference in notes (though likely that is too much detail to include):
SV: https://ci.adoptopenjdk.net/view/System%20tests/
OpenJDK regression: https://ci.adoptopenjdk.net/view/OpenJDK%20tests/job/openjdk8_j9_openjdktest_x86-64_linux/

File: doc/release-notes/0.8/0.8.md (hunk 9 above)
Comment (SueChaplain) [reply doc/release-notes/0.8/0.8.md]: Well spotted, thanks!

File: doc/release-notes/0.8/0.8.md (hunk 7 above)
Comment (SueChaplain) [reply doc/release-notes/0.8/0.8.md]: I think we need a static list so that there is a fixed reference point for test failures. I've made the release notes match the query above. We'll have to do a last minute review to ensure that we still line up at GA.

File: doc/release-notes/0.8/0.8.md (hunk 8 above)
Comment (SueChaplain) [reply doc/release-notes/0.8/0.8.md]: Done

File: doc/release-notes/0.8/0.8.md (hunk 10 above)
Comment (SueChaplain) [reply doc/release-notes/0.8/0.8.md]: Changes:

- Updated the first paragraph to point to the OpenJ9 release plan at Eclipse.
- Added a sentence to supported platforms to make it clearer that we are talking V8, not V9:
*OpenJDK Version 8 binaries containing the Eclipse OpenJ9 virtual machine are supported with this release.*

**Comment by SueChaplain** [comment]
Hopefully I haven't missed anything from the recent raft of comments. Please comment if you spot something.

File: doc/release-notes/0.8/0.8.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+These release notes support the [Eclipse OpenJ9 release plan](https://projects.eclipse.org/projects/technology.openj9/releases/0.8/plan).
+
+
+## Supported platforms
+
+OpenJDK Version 8 binaries containing the Eclipse OpenJ9 virtual machine are supported with this release.
+
+These binaries are available from the [AdoptOpenJDK community](https://adoptopenjdk.net/?variant=openjdk8-openj9).
+Builds are tested against the OpenJDK test suite.
```
Comment (DanHeidinga) [inline doc/release-notes/0.8/0.8.md]: Can we change this sentence to something like: `Builds are tested against the OpenJ9 FV test suite, the OpenJDK test suites, and additional tests at AdoptOpenJDK.`

I would rather put the emphasis on OpenJ9 and Adopt.

**Comment by pshipton** [comment]
FYI this new issue for the release notes https://github.com/eclipse/openj9/issues/1252

**Comment by SueChaplain** [comment]
Test statement updated:
*Builds are tested against the OpenJ9 FV test suite, the OpenJDK test suites, and additional tests at AdoptOpenJDK.*

**Comment by pshipton** [comment]
lgtm
